src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs

```diff
@@ -164,6 +165,288 @@ public static void XmlDictionaryWriter_InvalidUnicodeChar()
+        IXmlMtomWriterInitializer writerInitializer = (IXmlMtomWriterInitializer)binaryWriter;
+        writerInitializer.SetOutput(ms, Encoding.UTF8, int.MaxValue, "application/soap+xml", null, null, true, false);
+        serializer.WriteObject(ms, td);
+        binaryWriter.Flush();
+        byte[] xmlDoc = ms.ToArray();
+        binaryWriter.Close();
+        Encoding[] allEncodings = new Encoding[] { Encoding.Unicode, Encoding.BigEndianUnicode, Encoding.UTF8 };
+        XmlDictionaryReader mtomReader = XmlDictionaryReader.CreateMtomReader(xmlDoc, 0, xmlDoc.Length, allEncodings, null, XmlDictionaryReaderQuotas.Max, int.MaxValue, new OnXmlDictionaryReaderClose((XmlDictionaryReader reader) => { }));
+        IXmlMtomReaderInitializer readerInitializer = (IXmlMtomReaderInitializer)mtomReader;
+        readerInitializer.SetInput(xmlDoc, 0, xmlDoc.Length, allEncodings, null, XmlDictionaryReaderQuotas.Max, int.MaxValue, new OnXmlDictionaryReaderClose((XmlDictionaryReader reader) => { }));
+        var b = mtomReader.ReadContentAsObject();
+        mtomReader.Close();
+    }
+
+    [ActiveIssue(12902)]
+    [Fact]
+    public static void IXmlTextReaderInitializerTest()
+    {
+        DataContractSerializer serializer = new DataContractSerializer(typeof(TestData));
+        MemoryStream ms = new MemoryStream();
+        TestData td = new TestData();
+        Encoding encoding = Encoding.UTF8;
+        XmlDictionaryWriter textWriter = XmlDictionaryWriter.CreateTextWriter(ms, encoding, false);
+        IXmlTextWriterInitializer writerInitializer = (IXmlTextWriterInitializer)textWriter;
+        writerInitializer.SetOutput(ms, encoding, false);
+        serializer.WriteObject(ms, td);
+        textWriter.Flush();
+        byte[] xmlDoc = ms.ToArray();
+        textWriter.Close();
+        XmlDictionaryReader textReader = XmlDictionaryReader.CreateTextReader(xmlDoc, 0, xmlDoc.Length, encoding, XmlDictionaryReaderQuotas.Max, new OnXmlDictionaryReaderClose((XmlDictionaryReader reader) => { }));
+        IXmlTextReaderInitializer readerInitializer = (IXmlTextReaderInitializer)textReader;
+        readerInitializer.SetInput(xmlDoc, 0, xmlDoc.Length, encoding, XmlDictionaryReaderQuotas.Max, new OnXmlDictionaryReaderClose((XmlDictionaryReader reader) => { }));
+        textReader.ReadContentAsObject();
+        textReader.Close();
+    }
+
+    static bool ReadTest(MemoryStream ms, Encoding encoding, ReaderWriterFactory.ReaderWriterType rwType, byte[] byteArray)
+    {
+        ms.Position = 0;
+        XmlDictionaryReader reader = (XmlDictionaryReader)ReaderWriterFactory.CreateXmlReader(rwType, ms, encoding);
+        reader.ReadToDescendant("Root");
+        byte[] bytesFromReader = reader.ReadElementContentAsBase64();
+        if (bytesFromReader.Length != byteArray.Length)
+        {
+            Console.WriteLine("Error, different array lengths!");
```

Comment:
Please remove calls to Console.WriteLine or replace them with asserts throughout the PR.